src/base/http/connection.cpp

```diff
@@ -79,9 +76,6 @@ void Connection::read()
             return;
 
         case RequestParser::ParseStatus::BadRequest: {
-                Logger::instance()->addMessage(tr("Bad Http request, closing socket. IP: %1")
```

Comment:
IMO logging bad clients is sometimes useful, for example a user can know if there is someone messing the server with invalid/bad HTTP requests and setup a firewall to block it. 
The 501 error not worth logging is for another reason, it is a valid request unlike the aforementioned ones.
tldr; just implement the new code and leave the others.

Reply:
Done